src/arch/xtensa/task.c

```diff
@@ -176,6 +176,7 @@ int arch_run_task(struct task *task)
 
 int arch_allocate_tasks(void)
 {
+	unsigned int cpu = cpu_get_id();
```

Comment:
The CPU core for each task will only be known at task creation time. i.e. after topology has been loaded. So we cannot hard code to current CPU unless this is being run on ALL CPUs.

Reply:
@lgirdwood Sorry, where is this done? I see
```
        interrupt_register(PLATFORM_IRQ_TASK_LOW, IRQ_AUTO_UNMASK, _irq_task,
                           task_irq_low_get());
```
where PLATFORM_IRQ_TASK_LOW is IRQ_NUM_SOFTWARE[234] on all platforms, i.e. a DSP interrupt. Similar for MED and HIGH. That means they are registered on the current CPU. What am I missing?

Reply:
Not yet implemented on other cores atm, topology can request core > 0 but code will assign 0 (although this will change with upcoming PRs).

Reply:
hm, which PR is this? I think it should actually become better with these changes: instead of mangling on bitfields you'd call interrupt_enable() with a specific CPU, that's it. And it would be a good test case for that!

Reply:
PRs for scheduling pipelines on other cores are in progress by @tlauda @mmaka1 , I dont know when they will be ready for merging though.

Reply:
would it be possible to merge this first and then port their work on top?